### PR TITLE
gradio web ui

### PR DESCRIPTION
**Pull request by AK391**
created a web ui on gradio

### CODE REVIEW

File: demo.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        sys.exit(1)
+
+run_cmd("wget --continue https://github.com/intel-isl/DPT/releases/download/1_0/dpt_large-midas-2f21e586.pt -O weights/dpt_large-midas-2f21e586.pt")
+run_cmd("wget --continue https://github.com/intel-isl/DPT/releases/download/1_0/dpt_large-ade20k-b12dca68.pt -O weights/dpt_large-ade20k-b12dca68.pt")
+
+def process(img, inference_type):
+    _id = randint(1, 10000)
+    INPUT_DIR = "/tmp/input_image" + str(_id) + "/"
+    OUTPUT_DIR = "/tmp/output_image" + str(_id) + "/"
+    run_cmd("rm -rf " + INPUT_DIR)
```
Comment (ranftlr) [inline demo.py:28]: Executing `rm -rf` with paths that are constructed by a wep-app is very dangerous. I can't merge this with shell commands that are excecuted.

File: run_monodepth.py [hunk 2]
```diff
@@ -39,15 +39,15 @@ def run(input_path, output_path, model_path, model_type="dpt_hybrid", optimize=T
             path=model_path,
             backbone="vitl16_384",
             non_negative=True,
-            enable_attention_hooks=args.vis,
+            enable_attention_hooks=False, #args.vis,
```
Comment (ranftlr) [inline run_monodepth.py:42]: Any reason why this is hardcode in this PR?

**Review by ranftlr — CHANGES_REQUESTED** [review]
Thank you for your contribution. Unfortunately, we can't accept PRs that include file manipulations and shell commands (especially if accessible via the web).

However, I think this could be entirely avoided by actually integrating the model into the script, instead of calling the original demo scripts.

**Comment by AK391** [comment]
@ranftlr Hi, thanks for reviewing, the way the current code is setup would require to make more changes to pass in a pil image instead of a file path, is it possible to add an inference module to the readme without running the command line commands?

**Comment by ranftlr** [comment]
I think the best way would be to restructure this PR. The basic flow would would be 

1) Load the models outside the process function so that they remain in memory. An example of this strategy is in the gradio guide: https://www.gradio.app/ml_examples. Our models are up to 1.2 GB, loading them with every request makes every inference request much slower than it needs to be. Currently loading the model takes more time than processing the image. If total memory is an issue you could add a startup flag to pre-select the models that are loaded. 
2) Wrap inference code to do the appropriate input transform, pass to the right model, and perform any post-processing (up-sampling and creating of label overlays for segmentation, up-sampling and normalization for monodepth) that is necessary based on the model and task (see corresponding `run` functions that are currently called).
3) Create a function `process` that receives the PIL image and passes the image to the appropriate wrapper function.

The upside of this would be that the code would be (potentially much) faster and wouldn't require any shell commands.

**Pull request closed without merging**